### PR TITLE
CPM-615 add setFile user intent

### PR DESCRIPTION
**Pull request by amigaire**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [x] Tests

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php [hunk 1]
```diff
@@ -38,6 +39,7 @@ public function __construct(
         private ?SetEnabled $enabledUserIntent = null,
         private ?AssociationUserIntentCollection $associationUserIntents = null,
         private ?QuantifiedAssociationUserIntentCollection $quantifiedAssociationUserIntents = null,
+        private ?SetFile $fileUserIntent = null,
```
Comment (Balamung) [inline src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php:42]: Isn't the SetFile a ValueIntent ? :thinking:

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetFile.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class SetFile implements MediaUserIntent
+{
+    public function __construct(
+        private string $attributeCode,
+        private string $filePath,
```
Comment (Balamung) [inline src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetFile.php:17]: In other value intents, the value is the last param(s), can we move the filePath to be the last param as well ?

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php [hunk 3]
```diff
@@ -78,6 +81,9 @@ public static function createFromCollection(int $userId, string $productIdentifi
             } elseif ($userIntent instanceof ParentUserIntent) {
                 Assert::null($parentUserIntent, 'Only one parent intent can be passed to the command.');
                 $parentUserIntent = $userIntent;
+            } elseif ($userIntent instanceof SetFile) {
+                Assert::null($fileUserIntent, 'Only one file intent can be passed to the command.');
```
Comment (Balamung) [inline src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php:85]: It correlates with the rest, but we shouldn't limit this, as we should be able to edit multiple File attributes in one command :)

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php (hunk 1 above)
Comment (amigaire) [reply src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php]: I move it to a valueIntent

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/MediaUserIntent.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Product\API\Command\UserIntent;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface MediaUserIntent extends UserIntent
```
[the remark this answers is not in the document]
Comment (amigaire) [reply src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/MediaUserIntent.php]: this "middle" interface is deleted now

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php (hunk 3 above)
Comment (amigaire) [reply src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php]: no limit anymore

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetFileValue.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class SetFileValue implements ValueUserIntent
+{
+    public function __construct(
+        private string $attributeCode,
+        private ?string $channelCode,
+        private ?string $localeCode,
+        private string $value
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetFileValue.php:19]: `value` is too generic for me, and not helpful for someone who wants to use the API. What is it? file path? a filename? a key?

**Review by nmarniesse — APPROVED** [review]
GTM after fix

**Review by Balamung — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetFileValue.php (hunk 5 above)
Comment (amigaire) [reply src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetFileValue.php]: Changed for filePath

**Pull request merged**